Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -259,15 +333,54 @@ public class HTTPNetworkTransport: NetworkTransport {
     return request
   }
 
-  private func requestBody<Operation: GraphQLOperation>(for operation: Operation) -> GraphQLMap {
-    if sendOperationIdentifiers {
+  private func requestBody<Operation: GraphQLOperation>(for operation: Operation,
+                                                        sendQueryDocument: Bool,
+                                                        autoPersistQueries: Bool) -> GraphQLMap {
+    
+    var payload: GraphQLMap = [:]
+    
+    if autoPersistQueries {
       guard let operationIdentifier = operation.operationIdentifier else {
-        preconditionFailure("To send operation identifiers, Apollo types must be generated with operationIdentifiers")
+        preconditionFailure("To enabled autoPersistQueries, Apollo types must be generated with operationIdentifiers")
       }
+      payload["extensions"] = [
+        "persistedQuery" : ["sha256Hash": operationIdentifier, "version": 1]
+      ]
+    }
+    
+    if let variables = operation.variables?.compactMapValues({ $0 }), variables.count > 0 {
+      payload["variables"] = variables
+    }
+    
+    if sendQueryDocument {
+      // TODO: This work-around fix "operationId is invalid for swift codegen" (https://github.com/apollographql/apollo-tooling/issues/1362), please remove this work-around after it's fixed.
+      let modifiedQuery = operation.queryDocument.replacingOccurrences(of: "fragment", with: "\nfragment")
+      payload["query"] = modifiedQuery
+    }
+    
+    return payload
+  }
+}
+
+extension URLSession {
+  func synchronousDataTask(with request: URLRequest) -> (Data?, URLResponse?, Error?) {
```

Comment:
I'm not sure if I'm missing something here, but this does not appear to be used. Can we get rid of it?

Reply:
removed